implementations/rust/ockam/ockam_node/src/context/context_lifecycle.rs

```diff
@@ -119,7 +119,7 @@ impl Context {
             self.transports.clone(),
             &self.flow_controls,
             #[cfg(feature = "std")]
-            self.tracing_context(),
+            OpenTelemetryContext::current(),
```

Comment:
What's the rationale behind this change? I'm trying to understand when we need to get the "global" context or just copy the one available at `self`.

Reply:
That's all due to the complexity of:

 - Having to pass the current tracing context between async processes when working with our workers. We work around this by setting a tracing context on the worker Context
 
 - Having a different notion of "tracing context" between the tracing crate, which creates spans using the `#[instrument]` macro, and the opentelemetry crate which creates the span data sent to the collector at a different point in time (when exiting the instrumented function). It's also the `opentelemetry` crate which generates span ids. So in order to get the latest span id, the one corresponding to the span created by the latest `#[instrument]`ed method, we need to reactualize the OpenTelemetry context using all the code in `OpenTelemetryContext::current()`.